src/openms/source/ANALYSIS/MAPMATCHING/TransformationModel.cpp

```diff
@@ -60,5 +63,295 @@ namespace OpenMS
   {
     params.clear();
   }
+  
+  void TransformationModel::weightData(TransformationModel::DataPoints& data, const Param& params)
+  {
+    // weight x values 
+    std::vector<std::string> valid_weights;
+    valid_weights = getValidXWeights();
+    if (params.exists("x_weight") && checkValidWeight(params.getValue("x_weight"), valid_weights) && !data.empty())
+    {
+      std::string x_weight = params.getValue("x_weight");
+      x_weight_ = x_weight;
```

Comment:
or directly assign x_weight_ instead of x_weight above. that way you could remove this line